src/Classes/Hotel.js

```diff
@@ -1,8 +1,18 @@
+import domUpdates from "../domUpdates";
+
 class Hotel {
     constructor(rooms, bookings) {
         this.rooms = rooms;
         this.bookings = bookings;
     };
+
+    checkAvailability(dates) {
+        this.bookings.forEach(booking => {
+            let bookingDate = new Date(booking.date);
+            let requestedDate = new Date(dates);
```

Comment:
I'm not sure I track the logic here, would love to hear your thinking

Reply:
good question Daniel. I was thinking that based on a users date selection, I would pass that date into this function... then, I would iterate through the list of all bookings to match that date with any upcoming booking dates... from there I would either confirm the booking and do my POST, or "deny" the booking with some message. I haven't really figured that all out yet, but this was my starting point. Thanks for the feedback.